oxidation/crates/parsec_api_protocol/src/lib.rs

```diff
@@ -10,36 +11,24 @@ mod realm;
 mod user;
 mod vlob;
 
-pub use block::*;
-pub use events::*;
-pub use invite::*;
-pub use message::*;
-pub use organization::*;
-pub use ping::*;
-pub use realm::*;
-pub use user::*;
-pub use vlob::*;
+use block::*;
+pub use cmds::*;
+use events::*;
+use invite::*;
+use message::*;
+use organization::*;
+use ping::*;
+use realm::*;
+use user::*;
+use vlob::*;
 
-use serde::{Deserialize, Serialize};
-
-macro_rules! impl_api_protocol_dump_load {
-    ($name:ident) => {
-        impl $name {
-            pub fn dump(&self) -> Vec<u8> {
-                rmp_serde::to_vec_named(&self).unwrap_or_else(|_| unreachable!())
-            }
-
-            pub fn load(data: &[u8]) -> Result<Self, &'static str> {
-                rmp_serde::from_read_ref::<_, Self>(&data).map_err(|_| "Invalid data")
-            }
-        }
-    };
-}
-
-pub(crate) use impl_api_protocol_dump_load;
-
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-#[serde(rename_all = "lowercase")]
-pub enum Status {
-    Ok,
-}
+pub use events::APIEvent;
```

Comment:
What is the need for the `use xxx::*;` then `pub use xxx::{yyy, zzz}` pattern ?

is it to allow to have simpler use in the sub modules (e.g. can do `use crate::yyy` instead of `use crate::xxx::yyy`) ?

What's the advantage over the regular `pub use xxx::°` ?

Reply:
_is it to allow to have simpler use in the sub modules (e.g. can do use crate::yyy instead of use crate::xxx::yyy) ?_ - yes
_What's the advantage over the regular pub use xxx::° ?_ - It's easier to import as you said

Reply:
Considering:

```rust
// spam/mod.rs
use foo::*;
pub use foo::Public1;

// spam/foo.rs
struct Private1 {};
pub struct Public1 {};
pub struct Public2 {};
```

Here we have `Public2` this is us marked `pub`, but the caller has to do `spam::foo::Public2` to access it.
This might be a good thing (for instance if the API is divided between high level and a low level parts, this way low level stuff is in a sub module to ease readability)

However in our current case there is no need for this, so I don't see the value of separating `use` vs `pub use` over just doing `pub use foo::*` (given `pub use` only expose pub stuff anyway)